docs_src/microservices/application/Triggers.md

```diff
@@ -27,13 +31,98 @@ Type = 'zero' #specifies of message bus (i.e zero for ZMQ)
 
+!!! note
+When using MQTT for the message bus, there is additional configuration required for specifying the MQTT specific options. 
+
+Here is example `MessageBus` configuration when using MQTT as the message bus:
+
+```toml
+[MessageBus]
+Type = "mqtt"
+    [MessageBus.SubscribeHost]
+        Host = "localhost"
+        Port = 1883
+        Protocol = "tcp"
+    [MessageBus.PublishHost]
+        Host = "localhost"
+        Port = 1883
+        Protocol = "tcp"
+    [MessageBus.Optional]
+        # MQTT Specific options
+        Username =""
+        Password =""
+        ClientId ="AppService"
+        # Connection information
+        Qos          =  "0" # Quality of Sevice 0 (At most once), 1 (At least once) or 2 (Exactly once)
+        KeepAlive    =  "10" # Seconds (must be 2 or greater)
+        Retained     = "false"
+        AutoReconnect  = "true"
+        ConnectTimeout = "5" # Seconds
+        SkipCertVerify = "false"
+        CertFile       = ""
+        KeyFile        = ""
+        KeyPEMBlock    = ""
+        CertPEMBlock   = ""
+```
+
+!!! note
+The MQTT `MessageBus` implementation doesn't yet support retrieving secrets from the `Secret Store`. Thus the secrets for secure connection are currently in plain text in the configuration file.
+
+## MQTT Trigger
+
+A MQTT trigger will execute the pipeline every time data is received from the configured external MQTT broker on the configured subscribe topic.  
+
+!!! note
+The data received from the external MQTT broker is not wrapped with any metadata known to EdgeX. The data is handled as JSON or CBOR. The data is assumed to be JSON unless the first byte in the data is not a `{` , in which case it is then assumed to be CBOR.
+
+!!! note
+The data receive, encoded as JSON or CBOR, must match the `TargetType` define by your application service. The default  `TargetType` is an `Edgex Event`. See [TargetType](../AdvancedTopics/#target-type) for more details.
+
+### Type and Topic configuration 
+Here's an example:
+```toml
+Type="external-mqtt" 
+SubscribeTopic="some-request"
+PublishTopic=""
+```
+The `Type=` is set to `external-mqtt`. To receive data from the external MQTT Broker you must set your `SubscribeTopic=` to the appropriate topic that the external publisher is using. You may also designate a `PublishTopic=` if you wish to publish data back to the external MQTT Broker. `edgexcontext.Complete([]byte outputData)` - Will send data back to back to the external MQTT Broker on the topic specified by the `PublishTopic=` property
+
+### MQTT Broker configuration
+The other piece of configuration required are the MQTT Broker connection settings:
+```toml
+[MqttBrokerConfig]
```

Comment:
According to [app-functions-sdk PR#498](https://github.com/edgexfoundry/app-functions-sdk-go/pull/498/files#diff-ffdab67f2576beda8a51e3f321b4c8a4R57), the section name here should be [MqttBroker].

Reply:
Great Catch!!